docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,215 @@
+    labels:
+      app.kubernetes.io/managed-by: <OADP NonAdminController id>
+      openshift.io/oadp: 'True'
+  ```
+- **Velero runs Restore**: Velero executes the restore operation based on the configuration specified in the Velero Restore object. Velero updates the status of the Velero Restore object to reflect the outcome of the restore process.
+- **Reconcile loop updates NonAdminRestore object Status**: Upon detecting changes in the status of the Velero Restore object, the NonAdminRestore controller's reconciliation loop updates the Status field of the corresponding NonAdminRestore object with the updated status from the Velero Restore object.
+
+- // TODO: Diagram remaining
+
+#### Queuing mechanism on NAB/NAR CR status
+  - We will introduce a Queue status on NAB/NAR CR status in order to give some transparency on what is the current backup/restore request status or a general idea when it will get processed so that the non-admin users are not left to wonder about what's happening with their backup/restore.
+  - The queuing status would be added to the NAB/NAR CR status, and it would give a general idea about how many backup/restore request are still remaining to be processed before their own backup/restore operation gets triggered.
+  - For example, in a singular OADP:NAC environment, lets say there are 5 non-admin users sharing this single OADP/NAC/Velero instance and all of them triggered backups.
+    - Current Velero Backup statuses in this example are: Backup-1, Backup-2: Complete; Backup-3: InProgress; Backup-4, Backup-5: New
+    - Then Non-Admin Backup Status for Backup-1 and Backup-2 will have `QueuedStatus: Completed`
+    - Non-Admin Backup Status for Backup-3 will have `QueuedStatus: Processing`
+    - Non-Admin Backup Status for Backup-4 will have `QueuedStatus: 1 Processing`
+    - Non-Admin Backup Status for Backup-5 will have `QueuedStatus: 1 Processing, 1 New` Or `QueuedStatus: 2 Backups remaining`
+// TODO: Diagram remaining
+
+#### Validating Webhooks to Ensure more transparency and debug-ability
+Currently, the velero CLI is a very useful tool, particularly the describe backup option. This option comes in pretty handy to gain more detailed insights on what is getting backed up (ResourceList) or which operations are in progress (async datamover operations) or details like how many items are still remaining to be backed up and much more.
+We will be ensuring that non-admin users have view/read access to the Velero CRs that are only relevant to them or created by them.
+
+// TODO: Diagram remaining
+
+#### Control over the Velero spec exposed via Non-Admin CRs
+The Velero Backup/BSL/Restore CRs consists of a plethora of Backup Spec options. It might be desirable to have a control over the spec that is exposed or may be allowed listed for non-admin users.
+For instance, Backup Spec IncludedNamespaces should by default be the non-admin user's namespace as this is a scoped backup operation
```

Comment:
If user uses labelSelector option in that case it should only look for the resources in the resources which are part of non-admin user namespace, right ?

Reply:
I think its acceptable to include depended resources like VSC, PV etc which are cluster scoped.